### ISSUE TITLE
Generate Enum instances for enumerations

### ISSUE THREAD

**Issue by riz0id** [awakesecurity/proto3-suite#179]
Protocol buffer enumerations are not generated with `Enum` instances. They are generated with instances to `Bounded` and `Ord`, so there shouldn't be anything preventing `Enum` from also being provided by default.

**Comment by riz0id** [comment]
Right, being about to enumerate the set of fields for an enumeration would be convenient, e.g. 
```haskell
 [minBound @MyEnumeration .. maxBound @MyEnumeration]
 ```

**Comment by riz0id** [comment]
Although, it seems like the `Ord` instance respects field number, and having `Enum` deviate from that would be awkward?

**Comment by j6carey** [comment]
Would the `enumerate` method of type class `Proto3.Suite.Class.Finite` do what is needed?  Though that produces the codes, not the typed values.
